libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity/Scripts/Media/RemoteAudioSource.cs

```diff
@@ -58,14 +69,20 @@ public void Play()
             if (!IsPlaying)
             {
                 IsPlaying = true;
-                //PeerConnection.Peer.RemoteAudioFrameReady += RemoteAudioFrameReady;
+                OnAudioConfigurationChanged(deviceWasChanged: false);
+                _audioTrackReadBuffer = PeerConnection.Peer.CreateAudioTrackReadBuffer();
             }
         }
```

Comment:
@stephenatwork @djee-ms it feels weird that we can control play state on the linked AudioSource _and_ on this object independently. Looks like an easy source of mistakes. Also we need to decide a sane behavior for when the AudioSource plays but this doesn't (at the moment you get beeping).

I propose to
1. create/destroy the read buffer automatically when a track is added/removed
2. return empty frames in OnAudioFilterRead if there is no track
3. remove Play/Stop/IsPlaying from this class, or make them a shortcut for `_linkedAudioSource.(Play|Stop|IsPlaying)`

Thoughts?

Reply:
Sounds good to me. I suppose that there is a small overhead even when paused via the unity AudioSource, because IIRC the  buffering/transcoding still happens internally, even if we're not consuming the data via OnAudioFilterRead. Not sure if it's worth doing anything about that.

Reply:
At the moment we transcode only on read so we don't have this issue*. For the sake of correctness though I suppose we can override OnEnable/OnDisable to turn the buffer on/off.

*Maybe we _shouldn't_ wait for a Read() call (I am not sure if the processing adds any meaningful latency) but we can deal with this later.

Reply:
Yes, we should continue to transcode on read because this means transcoding inside the audio thread, which is the right place to do so, instead of some other thread like the WebRTC signaling thread or the main Unity app thread, which are busy doing other stuffs. And yes we should return silence when there's no track, to be consistent with the WebRTC behavior of sending silence when there's no track on a transceiver. I also agree on removing the playing state from here and use the audio source one only to avoid confusion.

Reply:
Filippo - when I say on read I mean OnAudioFilterRead. Do you mean something else? IIRC we're transcoding when the frame arrives from webrtc (addFrame), not when requested from OnAudioFilterRead (that part is a memcpy). The transcoding is the extra work I mean if the unity source is paused, but data continues to arrive from webrtc.